utils.py

```diff
@@ -721,6 +722,86 @@ def get_hashable_value(value):
         return value
 
 
+def convert_to_text_angular(html_data):
+    """This fucntion converts the html to text angular supported format.
+
+    Args:
+        html_data: str. HTML string to be converted.
+
+    Returns:
+        str. The converted HTML string.
+    """
+    soup = bs4.BeautifulSoup(html_data, 'html.parser')
+
+    tags_to_remove = ['div', 'span', 'code', 'table', 'td', 'tr', 'tbody']
+
+    for tag_name in tags_to_remove:
+        for tag in soup.findAll(tag_name):
+            tag.unwrap()
+
+    for br in soup.findAll('br'):
+        if br.parent.name not in ['b', 'i', 'li', 'p']:
+            br.wrap(soup.new_tag('p'))
+
+    tag_list = soup.findAll()
+    if not len(tag_list):
+        start_tag = None
+    else:
+        start_tag = '<' + soup.findAll()[0].name + '>'
```

Comment:
I don't quite understand this line. According to BS4 documentation, you can run things like `soup.find('a')` (you don't need the brackets). Why do we need the brackets here?

Reply:
This part checks if a html string is not wrapped in any tag. Since it is not possible to obtain the position of a tag using BeautifulSoup, exact tag is to be searched in the html string. So, the brackets are needed here. I have also added a comment for the same.